src/bindings/python/src/pyopenvino/utils/utils.cpp

```diff
@@ -449,8 +449,14 @@ ov::Any py_object_to_any(const py::object& py_obj) {
         return py::cast<ov::intel_auto::SchedulePolicy>(py_obj);
     } else if (py::isinstance<ov::hint::SchedulingCoreType>(py_obj)) {
         return py::cast<ov::hint::SchedulingCoreType>(py_obj);
-    } else if (py::isinstance<std::set<ov::hint::ModelDistributionPolicy>>(py_obj)) {
-        return py::cast<std::set<ov::hint::ModelDistributionPolicy>>(py_obj);
+    } else if (py::isinstance<py::set>(py_obj)) {
```

Comment:
I don't think that it's good change -- what if py_obj is set of string for example?

Reply:
I'd suggest to implement it like py::list in the [py_object_to_any ](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/utils/utils.cpp#L426) function and please, try to make [this](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/utils/utils.cpp#L27) function templated to reuse it for set. And there you may add check for `ov::hint::ModelDistributionPolicy`

Reply:
Ok, updated.